crates/host/src/wasmbus/mod.rs

```diff
@@ -2164,16 +2175,55 @@ impl Host {
         Ok(buf.into())
     }
 
-    #[allow(unused)] // TODO: Remove once implemented
-    #[instrument(skip(self, payload))]
-    async fn handle_claims(&self, payload: impl AsRef<[u8]>) -> anyhow::Result<Bytes> {
-        bail!("TODO")
+    #[instrument(skip(self))]
+    async fn handle_claims(&self) -> anyhow::Result<Bytes> {
+        // TODO: update control interface client to have a more specific type definition for
+        // GetClaimsResponse, so we can re-use it here. Currently it's Vec<HashMap<String, String>>
+        #[derive(Serialize)]
+        struct ClaimsResponse {
+            claims: Vec<StoredClaims>,
+        }
+
+        let claims: Vec<StoredClaims> = self.scan_latticedata(CLAIMS_PREFIX).await?;
+        let resp = ClaimsResponse { claims };
+        Ok(serde_json::to_vec(&resp)?.into())
     }
 
-    #[allow(unused)] // TODO: Remove once implemented
-    #[instrument(skip(self, payload))]
-    async fn handle_links(&self, payload: impl AsRef<[u8]>) -> anyhow::Result<Bytes> {
-        bail!("TODO")
+    #[instrument(skip(self))]
+    async fn handle_links(&self) -> anyhow::Result<Bytes> {
+        let links: Vec<LinkDefinition> = self.scan_latticedata(LINKDEF_PREFIX).await?;
+        let resp = LinkDefinitionList { links };
+        Ok(serde_json::to_vec(&resp)?.into())
+    }
+
+    async fn scan_latticedata<T: DeserializeOwned>(
+        &self,
+        key_prefix: &str,
+    ) -> anyhow::Result<Vec<T>> {
+        let filtered_keys = self
+            .data
+            .keys()
+            .await
+            .context("failed to scan lattice data keys")?
+            .try_filter(|key| futures::future::ready(key.starts_with(key_prefix)))
```

Comment:
```suggestion
            .try_filter(|key| async { key.starts_with(key_prefix) })
```

Reply:
I had fun with lifetime errors earlier. I started with that, but:
```
error: lifetime may not live long enough
    --> crates/host/src/wasmbus/mod.rs:2191:31
     |
2191 |             .try_filter(|key| async { key.starts_with(CLAIMS_PREFIX) })
     |                          ---- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
     |                          |  |
     |                          |  return type of closure `[async block@crates/host/src/wasmbus/mod.rs:2191:31: 2191:71]` contains a lifetime `'2`
     |                          has type `&'1 std::string::String````